src/components/TodoItem.tsx

```diff
@@ -0,0 +1,107 @@
+import { useContext, useState } from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../types/Todo';
+import { deleteTodo, updateTodos } from '../api/todos';
+import { TodoContext } from '../contexts/TodoContext';
+import { ErrorMessage } from '../types/ErrorMessage';
+import { EditTodoForm } from './EditTodoForm';
+
+interface Props {
+  todo: Todo,
+  isTempTodo?: boolean,
+}
+
+export const TodoItem: React.FC<Props> = ({ todo, isTempTodo }) => {
+  const [isEditMode, setIsEditMode] = useState(false);
+  const {
+    setErrorMessage,
+    setTodos,
+    updateTodoList,
+    idsToUpdate,
+    idsToChange,
+  } = useContext(TodoContext);
+
+  const handleDeleteTodo = (todoId: number) => {
+    idsToUpdate(todoId);
+
+    deleteTodo(todoId)
+      .then(() => setTodos(prevTodos => prevTodos
+        .filter(todoToFilter => todoToFilter.id !== todoId)))
+      .catch(() => setErrorMessage(ErrorMessage.FailedDeleteTodo))
+      .finally(() => {
+        idsToUpdate(null);
+      });
+  };
+
+  const handleUpdateStatus = (todoToStatusUpdate: Omit<Todo, 'userId'>) => {
+    const { completed, id } = todoToStatusUpdate;
+
+    idsToUpdate(id);
+
+    updateTodos({ ...todoToStatusUpdate, completed: !completed })
+      .then(() => {
+        updateTodoList({ ...todoToStatusUpdate, completed: !completed });
+      })
+      .catch(() => setErrorMessage(ErrorMessage.FailedUpdateTodo))
+      .finally(() => idsToUpdate(null));
```

Comment:
the same with update
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/91946036/58051499-524e-4756-b715-afa8742343df)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/91946036/30d22dcb-b165-4a2e-bf08-f834eb0f1839)